### PR TITLE
Enigma

### PR DESCRIPTION
**Pull request by lars-wu**
InfoLong:
As the Enigma, you get a random clue about the killer each meeting, depending on the setting, you may have to report the body to receive a clue. The more tasks you complete the preciser the clues get.

### CODE REVIEW

**Comment by Tommy-XL** [comment]
I will look at the code later and also in this version we will not accept new roles for now except #68

**Comment by Tommy-XL** [comment]
image

File: Resources/Data/en.json [hunk 1]
```diff
@@ -877,6 +879,7 @@
   "ChiefOfPoliceInfoLong": "(Crewmates):\nPlayers with swords can be recruited to join the sheriff's team to serve the crew, but players without swords cannot be recruited.\n note: only one recruitment opportunity",
   "NiceMiniInfoLong": "(Crewmates):\nAs a Nice Mini, you can't be killed until you grow up, and if you die or are evicted from the meeting before you grow up, everyone loses.",
   "SpyInfoLong": "(Crewmates):\nAs the Spy, when someone uses their kill button on you (any ability that is used through the kill button), you'll see their name in orange for a few seconds.\nNote: If a Crewmate used their ability on you, you'll also see them with an orange name!\nNote: If you have no ability uses left, you won't see orange names at all!\nNote: If the kill button interaction is blocked the player's cooldown will reset to 10s'",
+  "EnigmaInfoLong": "(Crewmates):\nAs the Enigma, you get a random clue about the killer each meeting, depending on the setting, you may have to report the body to receive a clue. The more tasks you complete the preciser the clues get.",
```
Comment (0xDrMoe) [inline Resources/Data/en.json:882]: preciser is not english, please edit to become "more precise"

File: Resources/Lang/en_US.json
Comment (0xDrMoe) [inline Resources/Lang/en_US.json:1]: This could have the Stage # condensed to Stage {0} rather than having three strings, could it not?

Comment (0xDrMoe) [reply Resources/Lang/en_US.json:1]: now that I'm looking at it, I feel like a lot of them can have condensed variables

**Review by 0xDrMoe — CHANGES_REQUESTED** [review]

File: Resources/Lang/en_US.json
Comment (lars-wu) [reply Resources/Lang/en_US.json:1]: Yes, it is possible. I think it comes down to personal preference on how you want to do it. I like to have separate strings for each one.

File: Roles/Crewmate/Enigma.cs [hunk 2]
```diff
@@ -0,0 +1,545 @@
...
+                    stage = 1;
+
+                var clues = EnigmaClues.Where(a => a.ClueStage <= stage &&
+                    !ShownClues[playerId].Any(b => b.EnigmaClueType == a.EnigmaClueType && b.ClueStage == a.ClueStage))
+                    .ToList();
+                if (clues.Count == 0) continue;
+                if (showStageClue && clues.Any(a => a.ClueStage == stage))
+                    clues = clues.Where(a => a.ClueStage == stage).ToList();
+
+                EnigmaClue clue = clues[rd.Next(0, clues.Count - 1)];
```
Comment (Tommy-XL) [inline Roles/Crewmate/Enigma.cs:130]: I'm not sure but `rd.Next()` can't select the last digit

image

**Review by Tommy-XL — APPROVED** [review]

File: Roles/Crewmate/Enigma.cs (hunk 2 above)
Comment (lars-wu) [reply Roles/Crewmate/Enigma.cs]: Thank you Tommy, I didn't know that. I have adjusted it.

**Pull request merged**